lucille-plugins/lucille-pinecone/src/main/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexer.java

```diff
@@ -95,52 +97,60 @@ protected void sendToIndex(List<Document> documents) throws IndexerException {
     List<Document> uploadList = new ArrayList<>();
     for (Document doc : documents) {
       if (isDeletion(doc)) {
-        // if doc is in uploadList and is now marked for deletion, then remove that document in uploadList and add it to deleteList
-        // if deleteByPrefix is true, also delete all instances of those in uploadList
-        if (deleteByPrefix) {
-          uploadList = uploadList.stream().filter(other -> !other.getId().startsWith(doc.getId())).collect(
-              Collectors.toList());
-        } else {
-          uploadList = uploadList.stream().filter(other -> !other.getId().equals(doc.getId())).collect(Collectors.toList());
-        }
+        // always add to deleteList;
+        // in the case where doc is in uploadList and is now marked for deletion, then remove that document in uploadList
+        uploadList.removeIf(x -> x.getId().equals(doc.getId()));
```

Comment:
How would a document marked for deletion ever end up in the upload list? Don't we only add documents to the uploadList if isDeletion() is false?